src/Core/Models/Data/AzureQueueMessage.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Bit.Core.Models.Data
+{
+    public class AzureQueueMessage<T> where T : class
+    {
+        public bool BlobBackedMessage => Message == null && Messages == null;
+        [JsonProperty]
+        public Guid MessageId { get; private set; }
+        public string BlobPath { get; set; } = null;
+        public T Message { get; set; } = null;
```

Comment:
Do we need to both have Message and Messages? Without knowing any gritty details it feels like we would be fine with only having a list?

Reply:
I agree. This is a little bit prematurely implementation to support the way that the `event` queue works. I'm not really sure why, but that queue allows for single messages as well as arrays.

This might be a good time to introduce that simplification if we end up going the event queue path as well. Otherwise, I'd rather keep this in since I think that refactor is pretty inevitable.